app/src/webview/java/org/mozilla/focus/webview/TrackingProtectionWebViewClient.java

```diff
@@ -102,19 +102,20 @@ public AmazonWebResourceResponse shouldInterceptRequest(final AmazonWebView view
             return new AmazonWebResourceResponse(null, null, null);
         }
 
-        final UrlMatcher matcher = getMatcher(view.getContext());
+        final UrlMatcher blockedSiteMatcher = getMatcher(view.getContext());
 
         // Don't block the main frame from being loaded. This also protects against cases where we
         // open a link that redirects to another app (e.g. to the play store).
-        final Uri pageUri = Uri.parse(currentPageURL);
-
-//        if ((!request.isForMainFrame()) &&
```

Comment:
I dug through some code and it looks like the isForMainFrame was added to prevent non-browser links (like market://) from crashing on this matcher check.

If you verify that it still works, then I'm okay with it, but otherwise I'd leave it in.

https://github.com/mozilla-mobile/focus-android/issues/106#issuecomment-281780919

Reply:
I can't use `isForMainFrame` because with this version of the webView we get a string, rather than the expected type we get in Focus.

That being said, I'll dig in on your research – thanks for doing that (good idea!)!

Reply:
> it looks like the isForMainFrame was added to prevent non-browser links (like market://) from crashing on this matcher check.

Looking at the linked issue, it looks like the http/http(s) checks were added to prevent crashing with malformed URLs like (`market:://`) but isForMainFrame is unrelated. #38 is open for fixing the http(s) check, which is out of scope of this issue (and we haven't seen any problems for it yet - maybe AmazonWebView is different because we're on a TV).